specs/core/0_beacon-chain.md

```diff
@@ -2397,7 +2324,8 @@ def process_attestation(state: BeaconState, attestation: Attestation) -> None:
         # the attestation is trying to create
         Crosslink(
             crosslink_data_root=attestation.data.crosslink_data_root,
-            epoch=slot_to_epoch(attestation.data.slot)
+            epoch=min(slot_to_epoch(attestation.data.slot),
```

Comment:
Should this be replaced with `assert slot_to_epoch(attestation.data.slot) <= attestation.data.previous_crosslink.epoch + MAX_CROSSLINK_EPOCHS`? The reason is that the newly created crosslink will have a mismatch between `crosslink_data_root` and `epoch`.

Reply:
No because `attestation.data.slot` is meant to reflect the _current_ slot, not the slot of the end of the crosslink it is creating. The epoch of the end of the crosslink is specified by `min(slot_to_epoch(attestation.data.slot), attestation.latest_crosslink.epoch + MAX_CROSSLINK_EPOCHS)` so we don't need a separate variable for it.